### PR TITLE
Unpacked pile ups are straggling

### LINKED ISSUE #1473: Unpacked vessels are off by one frame
Initially reported by @scimas, initially fused with #1416.
Especially visible in low orbit with a nearby (loaded) packed vessel in the vicinity (say about 300 m away). The vessel will appear to jump along its orbit with respect to the active vessel when going in and out of warp, which is due to the active vessel being packed and unpacked while the other vessel remains packed.

I suspect that what is happening is that unpacked vessels have their parts interpreted by KSP as being one frame in the past during `FixedUpdate`, and in the present in `WaitForFixedUpdate` (after the physics engine runs), whereas Principia assumes they are in the present in `FixedUpdate`, and in the future in `WaitForFixedUpdate`.

This requires looking at the `PileUp` logic again to reverse its timing with respect to the `Plugin`.

### PR DESCRIPTION
**Pull request by eggrobin**
Fix #1473.

### CODE REVIEW

File: ksp_plugin/pile_up.hpp [hunk 1]
```diff
@@ -88,6 +88,8 @@ class PileUp {
   // |DeformPileUpIfNeeded|.
   void NudgeParts() const;
 
+  Instant const& time();
```
Comment (pleroy) [inline ksp_plugin/pile_up.hpp]: Document this method, in particular wrt authoritative-ness.

File: ksp_plugin/plugin.cpp [hunk 2]
```diff
@@ -503,10 +505,10 @@ void Plugin::FreeVesselsAndPartsAndCollectPileUps() {
   // the same subset.
   for (auto const& pair : vessels_) {
     Vessel& vessel = *pair.second;
-    vessel.ForSomePart([this](Part& first_part) {
+    vessel.ForSomePart([Δt, this](Part& first_part) {
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Capture `this` before the other parameters (just like for method calls).

File: ksp_plugin/plugin.cpp [hunk 3]
```diff
@@ -1317,11 +1318,12 @@ void Plugin::AddPart(not_null<Vessel*> const vessel,
...
-                                         name,
-                                         mass,
-                                         degrees_of_freedom,
-                                         std::move(deletion_callback));
+  auto part = make_not_null_unique<Part>(
+      part_id,
+      name,
+      mass,
+      degrees_of_freedom,
+      std::move(deletion_callback));
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Revert this change.

File: ksp_plugin/plugin.hpp [hunk 4]
```diff
@@ -213,7 +214,7 @@ class Plugin {
   // Advances time on the pile ups to |t|, filling the tails of all parts up to
```
Comment (pleroy) [inline ksp_plugin/plugin.hpp]: Fix the comment.

File: ksp_plugin/plugin.hpp [hunk 5]
```diff
@@ -242,6 +243,8 @@ class Plugin {
   // |Planetarium.InverseRotAngle| is in degrees.
   virtual void AdvanceTime(Instant const& t, Angle const& planetarium_rotation);
 
+  virtual void CatchUpVessel(GUID const& vessel_guid);
```
Comment (pleroy) [inline ksp_plugin/plugin.hpp]: Comment.

File: serialization/journal.proto [hunk 6]
```diff
@@ -105,13 +105,23 @@ message AdvanceTime {
   optional In in = 1;
 }
 
-message AdvanceParts {
+message CatchUpLaggingVessels {
   extend Method {
-    optional AdvanceParts extension = 5120;
+    optional CatchUpLaggingVessels extension = 5120;
```
Comment (pleroy) [inline serialization/journal.proto]: I'd rather use a new tag here to avoid future confusion.

File: ksp_plugin/pile_up.hpp (hunk 1 above)
Comment (eggrobin) [inline ksp_plugin/pile_up.hpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 2 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 3 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin/plugin.hpp (hunk 4 above)
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: Done.

File: ksp_plugin/plugin.hpp (hunk 5 above)
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: Done.

File: serialization/journal.proto (hunk 6 above)
Comment (eggrobin) [inline serialization/journal.proto]: As discussed, it's fine.

File: journal/player_test.cpp [hunk 7]
```diff
@@ -106,25 +106,22 @@ TEST_F(PlayerTest, DISABLED_Debug) {
   LOG(ERROR) << "Last successful method out/return: \n"
               << player.last_method_out_return().DebugString();
 
-#if 0
+#if 1
```
Comment (pleroy) [inline journal/player_test.cpp]: 0
